src/main/java/com/movie_collection/bll/services/MovieService.java

```diff
@@ -1,12 +1,48 @@
 package com.movie_collection.bll.services;
 
+import com.google.inject.Inject;
+import com.movie_collection.be.Movie;
 import com.movie_collection.bll.services.interfaces.IMovieService;
+import com.movie_collection.dal.interfaces.IMovieDAO;
 
+import java.sql.SQLException;
 import java.util.List;
 
 public class MovieService implements IMovieService {
+    private final IMovieDAO movieDAO;
+
+    @Inject
+    public MovieService(IMovieDAO movieDAO) {
+        this.movieDAO = movieDAO;
+    }
+
+    @Override
+    public List<Movie> getAllMovies() throws SQLException {
+        return movieDAO.getAllMovies();
+    }
+
+    @Override
+    public int createMovie(Movie movie) throws SQLException {
+        return movieDAO.createMovie(movie);
+    }
+
+    @Override
+    public int updateMovie(Movie movie) throws SQLException {
```

Comment:
lets see some pros and cons on using the int idk for me its still seems to be sketchy to return number or idk maybe as we talk it does the same job its new area for me

Reply:
For me, it is more about the adaptability to check the number of rows from further up the chain
Lets say you are deleting a movie that has 2 categories you want the result to be 3 (2 deletions from linking table, 1 from Movie table)